### ISSUE TITLE
[BUG] <ul> did not correctly consider the "list-style" drawing "::marker"

### ISSUE THREAD

**Issue by Mapaler** [yorickshan/html2canvas-pro#21]
Please make sure you are testing with the latest [release of html2canvas-pro](https://github.com/yorickshan/html2canvas-pro/releases). 
Old versions are not supported and issues reported for them will be closed.

# Please follow the general troubleshooting steps first:

- [x] You are using the latest [version](https://github.com/yorickshan/html2canvas-pro/releases)
- [x] You are testing using the non-minified version of html2canvas-pro and checked any potential issues reported in the console

### Bug reports:

` <ul> list-style: none; ` or ` <ul> list-style: decimal; ` both are drawn with a small square.
![firefox_开发者工具_—_智龙急速阵型_—_httplocalhost8080PADDashFormation_0233](https://github.com/user-attachments/assets/5de89959-5287-4fce-8cad-07e31dbfb7cc)
![firefox_4000日1队全奖励_-_P D_1P_Formation_Maker_—_Firefox_Deve_0232](https://github.com/user-attachments/assets/b1a136d8-c65a-4779-affb-6d029516b4c3)

![xnviewmp_4000日1队全奖励_-_P D_1P_Formation_Maker(5) png_-_XnVie_0234](https://github.com/user-attachments/assets/d22e1773-06ef-4139-b7dd-fb94d438c671)

Add this setting may stop drawing `::markers`, but also not consider `list-style`.
```css
ul>li{
	display: inline-block;
}
```
![xnviewmp_4000日1队全奖励_-_P D_1P_Formation_Maker(6) png_-_XnVie_0235](https://github.com/user-attachments/assets/e1d25b6a-7d0d-4a67-a439-5e1014735c51)

### Specifications:

 * html2canvas-pro version tested with: 1.5.6
 * Browser & version: Firefox v129
 * Operating system: Win10

**Comment by yorickshan** [comment]
Currently, ::marker is not supported. I will look into adding support when I have time. Contributions via PRs are also welcome.

**Comment by Mapaler** [comment]
The most pressing bug at the moment is, `list-style: none;` drawn `::marker`.

**Comment by yorickshan** [comment]
Additionally, the current example is too complex with too many elements. Please simplify it using the following demo code as an example. This code was created while I was troubleshooting the -webkit-text-stroke issue:

```
<!DOCTYPE html>
<html>

<head>
  <style>
    ul {
      list-style-type: none;
    }

    li {
      color: red;
      font-size: 20px;
      -webkit-text-stroke: 1px blue;
    }
  </style>
</head>

<body>
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
    <li>Item 3</li>
  </ul>
  <button id="capture">Capture</button>
  <script src="https://cdn.jsdelivr.net/npm/html2canvas-pro@1.5.7/dist/html2canvas-pro.min.js"></script>
  <script>
    document.getElementById('capture').addEventListener('click', function () {
      html2canvas(document.body).then(function (canvas) {
        document.body.appendChild(canvas);
      });
    });
  </script>
</body>

</html>
```

**Comment by yorickshan** [comment]
Using the example below, I was unable to reproduce the issue. Please check your code again or provide a minimal example that can reproduce the issue.

```
<!DOCTYPE html>
<html>

<head>
  <style>
    ul {
      list-style-type: none;
    }

    li {
      color: red;
      font-size: 20px;
    }
  </style>
</head>

<body>
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
    <li>Item 3</li>
  </ul>
  <button id="capture">Capture</button>
  <script src="https://cdn.jsdelivr.net/npm/html2canvas-pro@1.5.8/dist/html2canvas-pro.min.js"></script>
  <script>
    document.getElementById('capture').addEventListener('click', function () {
      html2canvas(document.body).then(function (canvas) {
        document.body.appendChild(canvas);
      });
    });
  </script>
</body>

</html>
```

![image](https://github.com/user-attachments/assets/509ed122-be17-467e-84de-6b3e3a542704)

**Comment by Mapaler** [comment]
![firefox_Firefox_Developer_Edition_0316](https://github.com/user-attachments/assets/8fb7fa82-2b3f-40ba-b839-4794082b4294)
```html
<!DOCTYPE html>
<html>
<head>
  <style>
    ul {
      display: inline-flex;
      list-style-type: none;
    }

    li {
      color: red;
      font-size: 20px;
      -webkit-text-stroke: 1px blue;
    }

    canvas {
      border: 1px solid black;
    }

  </style>
  <!--<script src="node_modules/@html2canvas/html2canvas/dist/html2canvas.min.js"></script>-->
  <script src="node_modules/html2canvas-pro/dist/html2canvas-pro.min.js"></script>
</head>

<body>
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
    <li>Item 3</li>
  </ul>
  <button id="capture">Capture</button>
  <script>
    document.getElementById('capture').addEventListener('click', function () {
      [...document.body.querySelectorAll("canvas")].forEach(ca=>ca.remove());
      html2canvas(document.body).then(function (canvas) {
        document.body.appendChild(canvas);
      });
    });
  </script>
</body>
</html>
```

**Comment by yorickshan** [comment]
> ![firefox_Firefox_Developer_Edition_0316](https://private-user-images.githubusercontent.com/6565860/351674911-8fb7fa82-2b3f-40ba-b839-4794082b4294.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4NzE5MjUsIm5iZiI6MTcyMTg3MTYyNSwicGF0aCI6Ii82NTY1ODYwLzM1MTY3NDkxMS04ZmI3ZmE4Mi0yYjNmLTQwYmEtYjgzOS00Nzk0MDgyYjQyOTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjVUMDE0MDI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWY3ODU3ODIwMWY3NGQ5Y2FiYjliMGY5MmJlMWI2OWQwNmEyYjVkMDVmN2RmNmFlYjM1MzcxY2FkMTk5M2JkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.shm7JCmJazXikQcH0qp6v1TlVqqzHKeT_WGLe7fhwQM)
> 
> ```
> <!DOCTYPE html>
> <html>
> <head>
>   <style>
>     ul {
>       display: inline-flex;
>       list-style-type: none;
>     }
> 
>     li {
>       color: red;
>       font-size: 20px;
>       -webkit-text-stroke: 1px blue;
>     }
> 
>     canvas {
>       border: 1px solid black;
>     }
> 
>   </style>
>   
>   <script src="node_modules/html2canvas-pro/dist/html2canvas-pro.min.js"></script>
> </head>
> 
> <body>
>   <ul>
>     <li>Item 1</li>
>     <li>Item 2</li>
>     <li>Item 3</li>
>   </ul>
>   <button id="capture">Capture</button>
>   <script>
>     document.getElementById('capture').addEventListener('click', function () {
>       [...document.body.querySelectorAll("canvas")].forEach(ca=>ca.remove());
>       html2canvas(document.body).then(function (canvas) {
>         document.body.appendChild(canvas);
>       });
>     });
>   </script>
> </body>
> </html>
> ```

fixed in html2canvas-pro v1.5.8

![image](https://github.com/user-attachments/assets/de909a25-32d3-4878-b2da-296486dd1f0b)
<!DOCTYPE html>
<html>

<head>
  <style>
    ul {
      display: inline-flex;
      list-style-type: none;
    }

    li {
      color: red;
      font-size: 20px;
      -webkit-text-stroke: 1px blue;
    }

    canvas {
      border: 1px solid black;
    }
  </style>
  
  <script src="https://cdn.jsdelivr.net/npm/html2canvas-pro@1.5.8/dist/html2canvas-pro.min.js"></script>
</head>

<body>
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
    <li>Item 3</li>
  </ul>
  <button id="capture">Capture</button>
  <script>
    document.getElementById('capture').addEventListener('click', function () {
      [...document.body.querySelectorAll("canvas")].forEach(ca => ca.remove());
      html2canvas(document.body).then(function (canvas) {
        document.body.appendChild(canvas);
      });
    });
  </script>
</body>

</html>

**Comment by Mapaler** [comment]
> fixed in html2canvas-pro v1.5.8
这个帖子的问题是，火狐浏览器， inline-flex 下前面的 li 被绘制了 ::marker 没有解决。

![firefox_Firefox_Developer_Edition_0325](https://github.com/user-attachments/assets/2d4ec6be-74c0-414d-8a7b-09dc5c784ae4)

**Comment by yorickshan** [comment]
> > fixed in html2canvas-pro v1.5.8
> 
> 这个帖子的问题是，火狐浏览器， inline-flex 下前面的 li 被绘制了 ::marker 没有解决。
> 
> ![firefox_Firefox_Developer_Edition_0325](https://private-user-images.githubusercontent.com/6565860/352101899-2d4ec6be-74c0-414d-8a7b-09dc5c784ae4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5NjI2MzgsIm5iZiI6MTcyMTk2MjMzOCwicGF0aCI6Ii82NTY1ODYwLzM1MjEwMTg5OS0yZDRlYzZiZS03NGMwLTQxNGQtOGE3Yi0wOWRjNWM3ODRhZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjZUMDI1MjE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDkwYzNlMjUwMDdlZWVkYWFiMTY3ZDNhZDZkMDkxMWI0ZTczOGI1MGE0MzcyNzUxNmMxOGJjZjc3YjEzYWRjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1kos6RJM9fTct4U2_MpI-r6RF2cpNIC24UIAz4VU1bA)

提供最小化复现 demo

**Comment by Mapaler** [comment]
> > > fixed in html2canvas-pro v1.5.8
> > 
> > 这个帖子的问题是，火狐浏览器， inline-flex 下前面的 li 被绘制了 ::marker 没有解决。
> > 
> > ![firefox_Firefox_Developer_Edition_0325](https://private-user-images.githubusercontent.com/6565860/352101899-2d4ec6be-74c0-414d-8a7b-09dc5c784ae4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5NjI2MzgsIm5iZiI6MTcyMTk2MjMzOCwicGF0aCI6Ii82NTY1ODYwLzM1MjEwMTg5OS0yZDRlYzZiZS03NGMwLTQxNGQtOGE3Yi0wOWRjNWM3ODRhZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjZUMDI1MjE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDkwYzNlMjUwMDdlZWVkYWFiMTY3ZDNhZDZkMDkxMWI0ZTczOGI1MGE0MzcyNzUxNmMxOGJjZjc3YjEzYWRjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1kos6RJM9fTct4U2_MpI-r6RF2cpNIC24UIAz4VU1bA)
> 
> 提供最小化复现 demo

就是你自己这个demo啊，你到火狐试一下。
